.github/workflows/generate-release-script.yml

```diff
@@ -0,0 +1,35 @@
+name: Generate Release
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  generate_release:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+
+    - name: Set up environment
+      run: echo "TIMESTAMP=$(date -d '${{ github.event.head_commit.timestamp }}' +%Y)" >> $GITHUB_ENV
+
+    - name: Zip the repository
+      run: zip -r Wall-E.zip .
+
+    - name: Upload Release Asset
+      uses: softprops/action-gh-release@v0.1.15
+      with:
+        files: Wall-E.zip
+        tag_name: ${{ env.TIMESTAMP }}
+        name: Wall-E Release ${{ env.TIMESTAMP }}
+        body: This automated release contains the latest Wall-E source code with the submodules. It was triggered by following commit - ${{ github.event.head_commit.id }}
```

Comment:
So, this will be triggered after each commit?

Reply:
Yes

Reply:
We don't want it to be released after each commit as possibility of reverting is significant, maybe there should be some trigger word added in the commit to trigger the release

Reply:
We can instead shift to first getting all development commits on the `dev` branch and keep the main branch only for releases.

Reply:
Or, we can have a separate release branch.

Reply:
Aah, I think having dev branch is better and then just merge all changes to main